src/rust/receiver-mock/src/logs.rs

```diff
@@ -0,0 +1,74 @@
+use std::collections::HashMap;
+use std::net::IpAddr;
+
+#[derive(Clone)]
+pub struct LogStats {
+    pub count: u64,
+    pub bytes: u64,
+}
+
+#[derive(Clone)]
+pub struct LogRepository {
+    pub total: LogStats,
+    pub ipaddr_to_stats: HashMap<IpAddr, LogStats>,
+}
+
+impl LogRepository {
+    pub fn new() -> Self {
+        return Self {
+            total: LogStats { count: 0, bytes: 0 },
+            ipaddr_to_stats: HashMap::new(),
+        };
+    }
+
+    pub fn add_log_message(&mut self, body: String, ipaddr: IpAddr) {
+        self.total.count += 1;
+        self.total.bytes += body.len() as u64;
+        let stats = self
+            .ipaddr_to_stats
+            .entry(ipaddr)
+            .or_insert(LogStats { count: 0, bytes: 0 });
+        stats.count += 1;
+        stats.bytes += body.len() as u64;
+    }
+
+    pub fn get_stats_for_ipaddr(&self, ipaddr: IpAddr) -> LogStats {
+        return self
+            .ipaddr_to_stats
+            .get(&ipaddr)
+            .unwrap_or(&LogStats { count: 0, bytes: 0 })
```

Comment:
Shouldn't this use an `Option` instead of returning 0s?

Reply:
If you want an Option, you can just call `.get` on the hashmap directly. The point of this method is to get zeroes if the key doesn't exist, which makes logical sense. Though honestly, this method only exists to make tests easier.

Reply:
Got it, my argument was about not returning a `LogStats` record when there isn't one in the internal map. I'm fine with the current state of this. Just thought it would be clearer with an `Option`